libutils/string_lib.c

```diff
@@ -77,7 +77,7 @@ unsigned int StringHash(const char *str, unsigned int seed, unsigned int max)
     h ^= (h >> 11);
     h += (h << 15);
 
-    return (h & (max - 1));
+    return (h % max);
```

Comment:
Os sorry I misread the parens place.
You're right, and this means that i made the same mistake in my example code.
It seems that the algorithm is fine.

We had the problem with a setup of a few thousand of agents. We found different approaches, and the has was one of them.